modules/wallet/encrypt_test.go

```diff
@@ -284,3 +284,32 @@ func TestInitFromSeed(t *testing.T) {
 		t.Fatalf("wallet should have correct balance after loading seed: wanted %v, got %v", origBal, newBal)
 	}
 }
+
+// TestReencrypt tests that Reencrypt re-encrypts a wallet correctly.
+func TestReencrypt(t *testing.T) {
+	if testing.Short() {
+		t.SkipNow()
+	}
+
+	wt, err := createBlankWalletTester(t.Name())
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer wt.closeWt()
+
+	var masterKey crypto.TwofishKey
+	fastrand.Read(masterKey[:])
+	_, err = wt.wallet.Encrypt(masterKey)
+	if err != nil {
+		t.Error(err)
+	}
+
+	var encryptionKey crypto.TwofishKey
+	seed, err := wt.wallet.Reencrypt(encryptionKey)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	walletKey := crypto.TwofishKey(crypto.HashObject(seed))
+	postEncryptionTesting(wt.miner, wt.wallet, walletKey)
+}
```

Comment:
Ah man. And, it's a PITA, but _really_ should be spending money that was collect from expired file contract outputs as well. Which actually probably needs to be done from the API package.

I think for this release we can just drop a TODO in the API package for that much complexity. This package should probably do some sending and receiving within this package on each side of the re-encrypt (which means having multiple wallets), and then a TODO in the API package about an ecosystem test that reinits one of the wallets halfway. Or maybe that'd make more sense as an antfarm job.